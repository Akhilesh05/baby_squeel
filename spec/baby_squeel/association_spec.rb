require 'spec_helper'
require 'baby_squeel/association'
require 'shared_examples/table'

describe BabySqueel::Association do
  subject(:table) {
    BabySqueel::Association.new(
      BabySqueel::Relation.new(Author),
      Author.reflect_on_association(:posts)
    )
  }

  it_behaves_like 'a table'
end
