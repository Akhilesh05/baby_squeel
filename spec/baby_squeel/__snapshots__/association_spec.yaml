---
BabySqueel::Association#== generates SQL 1: '"posts"."author_id" = 42'
BabySqueel::Association#!= generates SQL 1: ("posts"."author_id" != 42)
'BabySqueel::Association#== generates SQL 1 (Active Record: v5.2)': '"posts"."author_id"
  = ?'
'BabySqueel::Association#!= generates SQL 1 (Active Record: v5.2)': '"posts"."author_id"
  != ?'
